src/shared/utilities/timeoutUtils.ts

```diff
@@ -77,3 +77,28 @@ export class Timeout {
         }
     }
 }
+
+/**
+ * Invokes `fn()` until it returns a truthy value.
+ *
+ * @param fn  Function whose result is checked
+ * @param opt.timeout  Timeout in ms (default: 5000)
+ * @param opt.interval  Interval in ms between fn() checks (default: 500)
+ *
+ * @returns Result of `fn()`, or `undefined` if timeout was reached.
+ */
+export async function waitUntil<T>(
+    fn: () => Promise<T>,
+    opt: { timeout: number; interval: number } = { timeout: 5000, interval: 500 }
```

Comment:
I'd rather see this as a `{ attempts: number; interval: number }` instead of in terms of a timeout. If you have something like 
```
await waitUntil(
    async () => {
        ...network call that consistently takes 500 ms to complete
    },
    { timeout: 2000, interval: 500 }
)
```
you'd expect the function to run twice within the span of 2000 ms. However, this doesn't have a counter on the total timeout, which means that you'll instead make 4 calls over the span of 3500 ms.

Reply:
There are valid use cases for both retry based on number of attempts, and time-box based retries. A time boxed retry seems more suitable for this PRs needs (trying to get to a clean state during a test). Iteration based retries could be added at a later time.

Reply:
Yeah, the main concern is that the timebox isn't actually a timebox. If given the option for both, both would be great!